modules/openy_features/openy_node/modules/openy_node_activity/src/ActivityBatchUpdate.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * OpenY Node Activity batch update class.
+ */
+
+namespace Drupal\openy_node_activity;
+
+class ActivityBatchUpdate {
+
+  /**
+   * Update batch operation.
+   *
+   * @param array $activity_ids
+   *   Activity nodes to update.
+   * @param bool $status
+   *   Whether to publish or unpublish. True means publish.
+   * @param array $context
+   *   Batch context.
+   */
+  public static function updateActivities($activity_ids, $status, &$context) {
+    // Setup.
+    if (empty($context['sandbox'])) {
+      $context['sandbox']['progress'] = 0;
+      $context['sandbox']['max'] = count($activity_ids);
+    }
+    $items_per_iteration = 50;
+
+    $batch_ids = array_slice($activity_ids, $context['sandbox']['progress'], $items_per_iteration);
+    $activities = \Drupal::entityTypeManager()->getStorage('node')->loadMultiple($batch_ids);
+
+    if ($status) {
```

Comment:
I think this could be simplified:
```php
foreach ($activities as $node) {
  if ($status && !$node->isPublished()) {
    $node->setPublished();
    $node->save();
  }
  elseif(!$status && $node->isPublished()) {
    $node->setUnPublished();
    $node->save();
  }
  $context['sandbox']['progress']++;
  $context['results'][] = $node->id();
}
```